packages/block-library/src/shortcode/index.php

```diff
@@ -21,15 +21,13 @@ function render_block_core_shortcode( $attributes, $content ) {
  * Registers the `core/shortcode` block on server.
  */
 function register_block_core_shortcode() {
+	$path     = gutenberg_dir_path() . 'packages/block-library/src/shortcode/block.json';
```

Comment:
@gziolo I don't think this would work when this makes it into core, but I'm not sure what's the right way to load this in PHP. Any tips? I see you did a similar thing in #14238, but it never shipped in core, so I'm not sure if it's a solved problem

Reply:
It's quite a task to coordinate both Gutenberg and WordPress core :)

PHP files without any change are copied to this location in WordPress core:

https://github.com/WordPress/wordpress-develop/tree/master/src/wp-includes/blocks

`block.json` files aren't copied over in both Gutenberg and WordPress core and with the current folder structure, I don't think it would be possible. 

All PHP are whitelisted in WP core in here:
https://github.com/WordPress/wordpress-develop/blob/14376be6e195ce9cc7f4bf237651504777b504fb/tools/webpack/packages.js#L97-L109

I think it wouldn't be that hard to change it to something else and include `block.json` files there. I guess the simplest approach would be to put PHP file and its JSON file in the same folder to make them easy to reference.

In Gutenberg, PHP files for blocks are copied to the build folder with this logic:

https://github.com/WordPress/gutenberg/blob/3caee37456b6ff42eb1ac1701236bd58fbab7e8c/webpack.config.js#L117-L151

@aduth can provide more details for the PHP changes applied in Gutenberg if you need it for anything.

Reply:
> It's quite a task to coordinate both Gutenberg and WordPress core :)

😄
_____
Thanks for the guidance @gziolo, I'll get to work on including the block.json files in those directories.

Reply:
@gziolo 

I went ahead and made those adjustments 
For Gutenberg the change is here:
https://github.com/WordPress/gutenberg/pull/19534/files#diff-11e9f7f953edc64ba14b0cc350ae7b9dR152-R158

Then in WordPress core, I made the changes here:
https://github.com/chipsnyder/wordpress-develop/blob/rnmobile/shortcode-support/tools/webpack/packages.js#L155-L163

and here: 
https://github.com/chipsnyder/wordpress-develop/blob/rnmobile/shortcode-support/tools/webpack/packages.js#L244

I haven't opened up a PR for the changes to WordPress core yet because I wanted to check to see if this was in line with what you were thinking. I also wasn't sure about the process for that on updating the block-library package then committing the change and so forth.

Reply:
Yes, this is more or less what's necessary to get started. There is this question whether the PHP file and `block.json` file should be put in their own folder:
-- `build/block-library/blocks`
   -- `shortcode`
     -- `index.php`
     -- `block.json`

Reply:
> should be put in their own folder

I'll let you direct me on that approach. If we go that route though I may take a more direct approach by adding what I need to an index.native.js file then come back and do this more holistic approach in another PR.

Reply:
Whatever works best for you. We still need what you shared sooner than later :)

Reply:
I'm thinking let's start with `shortcode.json` and `shortcode.php` then we can come back later and move them to their own folder if needed. That way I minimize the risk of failing to update any references as part of this PR.

Reply:
If `block.json` is renamed to `shortcode.json` then we'll have to replace the filename to read in the php file as well.

Reply:
> then we'll have to replace the filename to read in the php file as well.

Is this ( https://github.com/WordPress/gutenberg/pull/19534/files#diff-0272f9f4a31a060e383e4d3cf5d5c7feR24  ) the change you're thinking of?

Reply:
Ah yes, I missed that one

Reply:
> Ah yes, I missed that one

No problem 🙂

I opened a PR for the WordPress-develop changes https://github.com/WordPress/wordpress-develop/pull/131 
@gziolo or @koke Do either of you know the process for updating the block-library version there?
Should it be part of that PR or should I open a new one after this merges?

Reply:
`@wordpress/block-library` is updated from npm. The changes applied to Gutenberg are released with the plugin release every two weeks and then published to npm.